Modules/hmacmodule.c

```diff
@@ -0,0 +1,277 @@
+#ifndef Py_BUILD_CORE_BUILTIN
+#  define Py_BUILD_CORE_MODULE 1
+#endif
+
+#include "pyconfig.h"
+#include "Python.h"
+#include "hashlib.h"
+
+#include "_hacl/Hacl_HMAC.h"
+
+#include "clinic/hmacmodule.c.h"
+
+#define HACL_HMAC_COMPUTE_HASH(HACL_FUNCTION, DIGEST_SIZE, KEY, SRC)    \
+    do {                                                                \
+        unsigned char out[DIGEST_SIZE];                                 \
+        HACL_FUNCTION(                                                  \
+            out,                                                        \
+            (uint8_t *)KEY->buf, (uint32_t)KEY->len,                    \
+            (uint8_t *)SRC->buf, (uint32_t)SRC->len                     \
```

Comment:
is there a guarantee somewhere that the caller won't attempt to HMAC more than 4GB at once?

Reply:
At the moment... no =/ But actually I hit an issue when I was implementing HMAC objects themselves these past days (I'll make a comment for that).

Reply:
I've added some checks. For 1-shot implementation, we cannot do anything else except rejecting the inputs if their length does not fit on an `uint32_t`.